packages/coin-vet/src/vet/secp256k1.ts

```diff
@@ -0,0 +1,99 @@
+import { randomBytes } from 'crypto'
+import { ec as EC } from 'elliptic'
+
+const curve = new EC('secp256k1')
+
+const N = Buffer.from('fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141', 'hex')
+const ZERO = Buffer.alloc(32, 0)
+
+function isValidPrivateKey(key: Buffer) {
+    return Buffer.isBuffer(key) &&
+        key.length === 32 &&
+        !key.equals(ZERO) &&
+        key.compare(N) < 0
+}
+
+function isValidMessageHash(hash: Buffer) {
+    return Buffer.isBuffer(hash) && hash.length === 32
+}
+
+/** secp256k1 methods set */
+export namespace secp256k1 {
+    /**
+     * generate private key
+     * @param rng the optional random number generator, which exactly generates 32 random bytes
+     */
+    export function generatePrivateKey(rng?: () => Buffer) {
+        rng = rng || (() => randomBytes(32))
+        for (; ;) {
+            const privKey = rng()
+            if (isValidPrivateKey(privKey)) {
+                return privKey
+            }
+        }
+    }
+
+    /**
+     * derive public key(uncompressed) from private key
+     * @param privKey the private key
+     */
+    export function derivePublicKey(privKey: Buffer) {
+        if (!isValidPrivateKey(privKey)) {
+            throw new Error('invalid private key')
+        }
+        const keyPair = curve.keyFromPrivate(privKey)
+        return Buffer.from(keyPair.getPublic().encode('array', false) as any)
+    }
+
+    /**
+     * sign a message using elliptic curve algorithm on the curve secp256k1
+     * @param msgHash hash of message
+     * @param privKey serialized private key
+     */
+    export function sign(msgHash: Buffer, privKey: Buffer) {
```

Comment:
Why do we need to implement sign function here?

Reply:
We will clean up the codes and have it fixed in the coming commit, along with the new functions.

Reply:
We have cleaned up the changes. Please verify. Thanks